cmd/server/app/serve.go

```diff
@@ -103,6 +104,15 @@ var serveCmd = &cobra.Command{
 			return fmt.Errorf("unable to prepare authz client for run: %w", err)
 		}
 
+		kc, err := keycloak.NewKeyCloak("", cfg.Identity.Server)
```

Comment:
What is the significance of the `name` argument to `NewKeyCloak`, and why is it left blank here?

Reply:
From the definition of IdentityProvider:
```
	// String returns the name of the identity provider.  This should be a short
	// one-word string suitable for presentation.  As a special case, a _single_
	// provider may use the empty string as its name to act as a default / fallback
	// provider.
```

We're using KeyCloak as the default provider.  At some point in the future, I'd like to be able to add various machine identities as providers which can be granted access to the system without needing to have accounts in our primary IDP.  (For example, an AWS role, a GCP service account, or a GitHub Action -- these might be stored as `user:gha/repo:octo-org/octo-repo:environment:prod` as the user key in OpenFGA, for example.)

I'm not ready to commit to implementing one of these machine identity providers yet, but I wanted to put the foundations in place while I was doing this refactoring.

Currently, we only have the one default identity provider, which is Keycloak.  This PR doesn't switch over all our Keycloak identity usage yet, only puts a backwards-compatible shim over the Keycloak + OpenFGA combination.